streaming/src/main/scala/org/apache/spark/streaming/ui/StreamingJobProgressListener.scala

```diff
@@ -70,9 +70,8 @@ private[streaming] class StreamingJobProgressListener(ssc: StreamingContext)
     runningBatchInfos(batchStarted.batchInfo.batchTime) = batchStarted.batchInfo
     waitingBatchInfos.remove(batchStarted.batchInfo.batchTime)
 
-    batchStarted.batchInfo.receivedBlockInfo.foreach { case (_, infos) =>
-      totalReceivedRecords += infos.map(_.numRecords).sum
-    }
+    // TODO. this should be fixed when input stream is not receiver based stream.
+    totalReceivedRecords += batchStarted.batchInfo.streamIdToNumRecords.values.sum
```

Comment:
This can be replaced by `batchStarted.batchInfo.numRecords` if you implement `numRecords` as I said above.